etc/cyclonedds.xsd

```diff
@@ -17,18 +17,18 @@ CycloneDDS configuration</xs:documentation>
 &lt;p&gt;The General element specifying Domain related settings.&lt;/p&gt;</xs:documentation>
     </xs:annotation>
     <xs:complexType>
-      <xs:choice minOccurs="0" maxOccurs="unbounded">
+      <xs:choice minOccurs="0">
```

Comment:
This seems to not be quite right: VS Code with the RedHat XML extensions (the version I have anyway) complain that a file with more than one element under Domain is invalid, and it seems to match the code here. I tried hacking it:
```diff
diff --git a/etc/cyclonedds.xsd b/etc/cyclonedds.xsd
index d38f2ea4..99558180 100644
--- a/etc/cyclonedds.xsd
+++ b/etc/cyclonedds.xsd
@@ -17,17 +17,17 @@ CycloneDDS configuration</xs:documentation>
 &lt;p&gt;The General element specifying Domain related settings.&lt;/p&gt;</xs:documentation>
     </xs:annotation>
     <xs:complexType>
-      <xs:choice minOccurs="0">
-        <xs:element ref="config:Compatibility"/>
-        <xs:element ref="config:Discovery"/>
-        <xs:element ref="config:General"/>
-        <xs:element ref="config:Internal"/>
-        <xs:element ref="config:Partitioning"/>
-        <xs:element ref="config:SSL"/>
-        <xs:element maxOccurs="unbounded" ref="config:Security"/>
-        <xs:element ref="config:Sizing"/>
-        <xs:element ref="config:TCP"/>
-        <xs:element ref="config:ThreadPool"/>
+      <xs:all>
+        <xs:element minOccurs="0" ref="config:Compatibility"/>
+        <xs:element minOccurs="0" ref="config:Discovery"/>
+        <xs:element minOccurs="0" ref="config:General"/>
+        <xs:element minOccurs="0" ref="config:Internal"/>
+        <xs:element minOccurs="0" ref="config:Partitioning"/>
+        <xs:element minOccurs="0" ref="config:SSL"/>
+        <xs:element minOccurs="0" ref="config:Security"/>
+        <xs:element minOccurs="0" ref="config:Sizing"/>
+        <xs:element minOccurs="0" ref="config:TCP"/>
+        <xs:element minOccurs="0" ref="config:ThreadPool"/>
         <xs:element minOccurs="0" name="Threads">
           <xs:annotation>
             <xs:documentation>
@@ -40,7 +40,7 @@ CycloneDDS configuration</xs:documentation>
           </xs:complexType>
         </xs:element>
         <xs:element ref="config:Tracing"/>
-      </xs:choice>
+      </xs:all>
       <xs:attribute name="Id">
         <xs:annotation>
           <xs:documentation>
```
and then this problem goes away. So I the use of `any`s needs to be a bit more liberal.

Reply:
Strictly speaking, according to the multiplicity, the `Security` group can occur "unbounded" and `all` only allows elements that do not occur or only occur once. I'll use the xsd spec and correct it. If this is correct, does that mean that the `Security` group may appear just once?

Reply:
I can live with it whichever way it goes, as long as all reasonable configuration files are valid according to the XSD. What I noticed is that one of those "reasonable" ones triggered a warning.